pruntime/enclave/src/lib.rs

```diff
@@ -1092,6 +1112,41 @@ fn sync_block(input: SyncBlockReq) -> Result<Value, Value> {
     }))
 }
 
+fn parse_events(block_with_events: &BlockWithEvents) -> Result<(), Value>{
+	let mut state = STATE.lock().unwrap();
+	let events = block_with_events.clone().events.unwrap();
+	let proof = block_with_events.clone().proof.unwrap();
+	let key = block_with_events.clone().key.unwrap();
```

Comment:
STATE is ok but don't unwrap the fields in inputs. Should return an error when these fields is None. Something like this:

```rust
let missing_field = err_msg("Missing field");
let events = block_with_events.clone().events.ok_or(missing_field)?;
...
```